lib/IRGen/GenFunc.cpp

```diff
@@ -1434,7 +1434,12 @@ void irgen::emitFunctionPartialApplication(
     llvm::Value *ctx = args.claimNext();
     if (isIndirectFormalParameter(*singleRefcountedConvention))
       ctx = IGF.Builder.CreateLoad(ctx, IGF.IGM.getPointerAlignment());
-    ctx = IGF.Builder.CreateBitCast(ctx, IGF.IGM.RefCountedPtrTy);
+    // We might get a struct containing a pointer e.g type <{ %AClass* }>
+    if (!ctx->getType()->isPointerTy())
+      ctx = IGF.coerceValue(ctx, IGF.IGM.RefCountedPtrTy,
+                            IGF.IGM.DataLayout);
+    else
+      ctx = IGF.Builder.CreateBitCast(ctx, IGF.IGM.RefCountedPtrTy);
```

Comment:
Wouldn't that case also be handled in coerceValue?

Reply:
Yes and applied this fix on master here: https://github.com/apple/swift/pull/21207